src/s3path.jl

```diff
@@ -95,7 +107,8 @@ function Base.tryparse(::Type{<:S3Path}, str::AbstractString; config::Union{Noth
     path = ()
     isdirectory = true
 
-    tokenized = split(str, "/")
+    version_split = split(str, "?version=")
```

Comment:
Let's stick to using `versionId` to match https://docs.aws.amazon.com/AmazonS3/latest/userguide/RetrievingObjectVersions.html#retrieve-obj-version-rest
```suggestion
    version_split = split(str, "?versionId=")
```
The parsing is frail and should probably use URIs.jl but I think it's adequate for the immediate future

Reply:
...I'm going to revert this chunk, I didn't actually intend to push this change on this branch! Will open in a separate PR, as originally intended (and will update to `versionId` there).

Reply:
Sounds great. Definitely not a critical feature for this PR and makes sense to be separate